### ISSUE TITLE
RHEL VM registration behind proxy does not work. 

### ISSUE THREAD

**Issue by LalatenduMohanty** [minishift/minishift#429]
The RHEL registration ( i.e .using subscription-manager) is not using the proxy flags (`$ minishift start --http-proxy <>`)
Currently we do use environment variable http_proxy in the RHEL virtual machine for registration and unregister and it does not work. 

[root@minishift ~]# cat /etc/profile.d/proxy.sh 
export http_proxy=http://fedora:fedora123@10.70.49.109:3128 
[root@minishift ~]# echo $http_proxy 
http://fedora:fedora123@10.70.49.109:3128

I manually tested the using following command from inside the VM to reproduce https://issues.jboss.org/browse/CDK-85 and  observed following. 

$ subscription-manager register --username=***** --password=***** --auto-attach --proxy=10.70.49.109:3128 --proxyuser=fedora --proxypassword=fedora123

I can see below logs in proxy server access log. 

```

1487337801.432   5689 10.10.60.253 TCP_TUNNEL/200 6848 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337805.528   3474 10.10.60.253 TCP_TUNNEL/200 7595 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337815.351   9206 10.10.60.253 TCP_TUNNEL/200 26040 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337819.556   3585 10.10.60.253 TCP_TUNNEL/200 7595 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337823.754   3583 10.10.60.253 TCP_TUNNEL/200 6635 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337827.953   3584 10.10.60.253 TCP_TUNNEL/200 7595 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337833.700   4505 10.10.60.253 TCP_TUNNEL/200 6347 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337843.826   9606 10.10.60.253 TCP_TUNNEL/200 39286 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337848.330   3856 10.10.60.253 TCP_TUNNEL/200 6592 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337853.002   4058 10.10.60.253 TCP_TUNNEL/200 32231 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337856.746   3193 10.10.60.253 TCP_TUNNEL/200 6560 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337860.931   3588 10.10.60.253 TCP_TUNNEL/200 6576 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337866.782   5160 10.10.60.253 TCP_TUNNEL/200 112987 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -
1487337871.579   4201 10.10.60.253 TCP_TUNNEL/200 27336 CONNECT subscription.rhsm.redhat.com:443 fedora HIER_DIRECT/10.4.164.72 -

```

However when I tried with `$ minishift start --http-proxy <>` I do not see such logs related to subscription.rhsm.redhat.com in access log but only found logs around accessing the Red Hat registry. 

```
1487362706.477   3547 10.10.60.253 TCP_TUNNEL/200 4383 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487362710.357   2723 10.10.60.253 TCP_TUNNEL/200 4383 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487362714.249   2725 10.10.60.253 TCP_TUNNEL/200 4998 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487362719.773   4352 10.10.60.253 TCP_TUNNEL/200 7475 CONNECT access.redhat.com:443 fedora HIER_DIRECT/104.92.7.131 -
1487362723.667   2722 10.10.60.253 TCP_TUNNEL/200 5187 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487362727.610   2764 10.10.60.253 TCP_TUNNEL/200 7513 CONNECT access.redhat.com:443 fedora HIER_DIRECT/104.92.7.131 -
1487362731.449   2720 10.10.60.253 TCP_TUNNEL/200 5175 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487362731.449   2720 10.10.60.253 TCP_TUNNEL/200 5175 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487362731.449   2720 10.10.60.253 TCP_TUNNEL/200 5175 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487362735.622   2998 10.10.60.253 TCP_TUNNEL/200 6309 CONNECT access.redhat.com:443 fedora HIER_DIRECT/104.92.7.131 -
1487362739.432   2764 10.10.60.253 TCP_TUNNEL/200 5175 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487363819.833 1087208 10.10.60.253 TCP_TUNNEL/200 10926878 CONNECT access.redhat.com:443 fedora HIER_DIRECT/104.92.7.131 -
1487363819.833 1079228 10.10.60.253 TCP_TUNNEL/200 3378769 CONNECT access.redhat.com:443 fedora HIER_DIRECT/104.92.7.131 -
1487363819.833 1087207 10.10.60.253 TCP_TUNNEL/200 11324432 CONNECT access.redhat.com:443 fedora HIER_DIRECT/104.92.7.131 -
1487364077.702      0 10.34.0.251 TCP_DENIED/407 4050 CONNECT vortex-win.data.microsoft.com:443 - HIER_NONE/- text/html
1487364080.701      0 10.34.0.251 TCP_DENIED/407 4058 CONNECT settings-win.data.microsoft.com:443 - HIER_NONE/- text/html
1487364081.393      0 10.34.0.251 TCP_DENIED/407 4058 CONNECT settings-win.data.microsoft.com:443 - HIER_NONE/- text/html
1487364140.898   3750 10.10.60.253 TCP_TUNNEL/200 5175 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487364142.755   2607 10.10.60.253 TCP_TUNNEL/200 5175 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487364143.378   2720 10.10.60.253 TCP_TUNNEL/200 5175 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
1487365689.925   3948 10.10.60.253 TCP_TUNNEL/200 4546 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487365695.243   2705 10.10.60.253 TCP_TUNNEL/200 5031 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487365695.878   2732 10.10.60.253 TCP_TUNNEL/200 4383 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487365699.758   2698 10.10.60.253 TCP_TUNNEL/200 5031 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487365702.321  78600 10.10.60.253 TCP_TUNNEL/200 5851857 CONNECT access.redhat.com:443 fedora HIER_DIRECT/104.74.26.11 -
1487365705.080   2758 10.10.60.253 TCP_TUNNEL/200 4546 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487365705.388   7728 10.10.60.253 TCP_TUNNEL/200 4546 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487365710.199   2643 10.10.60.253 TCP_TUNNEL/200 5022 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
1487365715.174   2674 10.10.60.253 TCP_TUNNEL/200 4543 CONNECT registry.access.redhat.com:443 fedora HIER_DIRECT/10.4.204.31 -
```

**Comment by LalatenduMohanty** [comment]
My idea is to introduce a new function (RhsmConfig() ) to set the subscription-manager for proxy environments (see below command). Once this config is set , all commands including registration/unregistration will use proxy. 
```
$ subscription-manager config --server.proxy_hostname=10.70.49.109 --server.proxy_port=3128 --server.proxy_user=fedora --server.proxy_password=fedora123
```

**Comment by LalatenduMohanty** [comment]
Also as per the RHEL documentation [1] http proxy environment variable is not supported for subscription manager. 

[1] https://access.redhat.com/documentation/en-US/Red_Hat_Subscription_Management/1/html-single/RHSM/index.html#rhsm-http-proxy

**Comment by hferentschik** [comment]
Merged via pull request #431
